psh/rmdir/rmdir.c

```diff
@@ -0,0 +1,103 @@
+/*
+ * Phoenix-RTOS
+ *
+ * rmdir - remove empty directories
+ *
+ * Copyright 2023 Phoenix Systems
+ * Author: Gerard Swiderski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+
+#include "../psh.h"
+
+
+static void psh_rmdir_info(void)
+{
+	printf("remove empty directories");
+}
+
+
+static void psh_rmdir_usage(void)
+{
+	printf("Usage: rmdir [-p] DIRECTORY...\n");
+}
+
+
+static int removedir(char *name, int parents)
+{
+	char *end = name + strlen(name) - (size_t)1;
```

Comment:
a lot of these `strlen()`
- `strdup()`
- this function
- `resolve_path()`

Reply:
Btw. this is the lesser of the two evils, the compiled binary of `psh` is smaller than e.g. after breaking `strdup()` into `strlen()`, `malloc()`, any kind of coping and pass the previously counted length in place of `strlen()` to this function, either not using `strlen()` at all and implementing it directly as a `while` loop, increases the size of `psh` binary (no good). For example, `removedir("/abc/def/ghi", 1)` will remove the empty directories: `/abc/def/ghi`, `/abc/def` and `/abc`, so there is no way to use only `resolve_path( )` instead.

Reply:
Maybe it's time for an upgrade in terms of speed for the `strlen()` function.

Reply:
I'll stay with `strlen()` for now ([why?](https://github.com/phoenix-rtos/phoenix-rtos-utils/pull/183#discussion_r1259527544)).